internal/controller/streams.go

```diff
@@ -196,21 +198,48 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 		logger.Error("Failed to send cluster metadata", zap.Error(err))
 		return err
 	}
+	var allWatchInfos []watcherInfo
+
+	// PHASE 1: List all resources
 	for resource, apiGroup := range resourceAPIGroupMap {
-		allResourcesSnapshotted.Add(1)
-		go resourceLister.DynamicListAndWatchResources(ctx, cancel, resource, apiGroup, &allResourcesSnapshotted, &snapshotCompleted)
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		resourceVersion, err := resourceLister.DynamicListResources(ctx, resourceLister.logger, resource, apiGroup)
+		if err != nil {
+			resourceLister.logger.Error("Failed to list resource", zap.String("resource", resource), zap.Error(err))
+			return err
+		}
+
+		allWatchInfos = append(allWatchInfos, watcherInfo{
+			resource:        resource,
+			apiGroup:        apiGroup,
+			resourceVersion: resourceVersion,
+		})
 	}
-	allResourcesSnapshotted.Wait()
+
+	// PHASE 2: Send snapshot complete
 	err = sm.sendResourceSnapshotComplete(logger)
+	if err != nil {
+		resourceLister.logger.Error("Failed to send snapshot complete", zap.Error(err))
+		return err
+	}
+	logger.Info("cloud-operator has successfully ingested and sent a snapshot of k8s cluster")
```

Comment:
We know that this is cloud-operator. No need to repeat it.
```suggestion
	logger.Info("Successfully sent resource snapshot")
```